pkg/operator/operator.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"context"
+	"crypto/x509"
+	"flag"
+	"io/ioutil"
+	"os"
+	"time"
+	"fmt"
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	nodeobservabilityv1alpha1 "github.com/openshift/node-observability-operator/api/v1alpha1"
+	nodeobservabilityv1alpha2 "github.com/openshift/node-observability-operator/api/v1alpha2"
+	operatorconfig "github.com/openshift/node-observability-operator/pkg/operator/config"
+	caconfigmapcontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/ca-configmap"
+	machineconfigcontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/machineconfig"
+	nodeobservabilityrun "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservabilityrun"
+	nodeobservabilitycontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservability"
+)
+
+const (
+	operatorName = "node_observability_operator"
+)
+
+var (
+	setupLog = ctrl.Log.WithName("operator")
+)
+
+// Client holds the API clients required by Operator.
+type Client struct {
+	client.Client
+	meta.RESTMapper
+}
+
+// Operator hold the client, manager, and logging resources
+// for the nodeobservability opreator.
+type Operator struct {
+	client  Client
+	manager manager.Manager
+	log     logr.Logger
+}
```

Comment:
It's a good idea to put all in  `operator.go` but let's make a dedicated PR for it: config can be combined with this.